tests/ui/span/issue-34264.stderr

```diff
@@ -1,8 +1,8 @@
-error: expected one of `:`, `@`, or `|`, found `<`
+error: expected one of `!`, `(`, `...`, `..=`, `..`, `::`, `:`, `{`, or `|`, found `<`
   --> $DIR/issue-34264.rs:1:14
    |
 LL | fn foo(Option<i32>, String) {}
-   |              ^ expected one of `:`, `@`, or `|`
+   |              ^ expected one of 9 possible tokens
    |
    = note: anonymous parameters are removed in the 2018 edition (see RFC 1685)
 help: if this is a `self` type, give it a parameter name
```

Comment:
Oh, I see that we already provide a bad arbitrary self type suggestion :-/

Reply:
Yes, and I think we can't just emit suggestions for pointer types in the parsing stage, since we can use type alias `type P<T> = Box<T>`. So I think this is acceptable and the current behaviour extends the old arbitrary self type suggestions.